libs/@local/hash-validation/src/data_type.rs

```diff
@@ -0,0 +1,279 @@
+use core::fmt;
+use std::{borrow::Borrow, fmt::Formatter};
+
+use error_stack::{bail, ensure, Report, ResultExt};
+use serde_json::Value as JsonValue;
+use thiserror::Error;
+use type_system::{url::VersionedUrl, DataType, DataTypeReference};
+
+use crate::{
+    error::{Actual, Expected},
+    OntologyTypeProvider, Schema, Validate,
+};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum JsonValueType {
```

Comment:
would more likely like this to be `JsonSchemaValueType` or something similar, `Integer` is not a valid data type in JSON itself. (or remove integer)